src/main/java/org/radarbase/appserver/config/MultiHttpSecurityConfig.java

```diff
@@ -35,78 +36,120 @@
+
+        // ensure the passwords are encoded properly
+        PasswordEncoder encoder = PasswordEncoderFactories.createDelegatingPasswordEncoder();
+
+        User.UserBuilder users = User.builder().passwordEncoder(encoder::encode);
+        InMemoryUserDetailsManager manager = new InMemoryUserDetailsManager();
+        manager.createUser(
+                users
+                        .username(adminUsername)
+                        .password(adminPassword)
+                        .roles("ADMIN")
+                        .authorities("ROLE_SYS_ADMIN")
+                        .build());
+        // manager.createUser(users.username("default").password("radar").roles("USER","ADMIN").build());
+        return manager;
     }
-  }
-
-  @Configuration
-  public static class WebSecurityConfig extends WebSecurityConfigurerAdapter {
-
-    @Override
-    protected void configure(HttpSecurity http) throws Exception {
-      // Allow all actuator endpoints.
-      http.requestMatcher(EndpointRequest.toAnyEndpoint())
-          .authorizeRequests()
-          .anyRequest()
-          .permitAll()
-          .and()
-          .csrf()
-          .disable();
+
+    @Configuration
+    @Order(1)
+    public static class AdminWebSecurityConfigurationAdapter extends WebSecurityConfigurerAdapter {
+
+        @Override
+        protected void configure(HttpSecurity http) throws Exception {
+
+            http.antMatcher("/v3/api-docs**")
+                    .authorizeRequests()
+                    .anyRequest()
+                    .permitAll()
+                    // .hasRole("ADMIN")
+                    .and()
+                    .httpBasic();
+        }
+    }
+
+    @Configuration
+    public static class WebSecurityConfig extends WebSecurityConfigurerAdapter {
+
+        @Override
+        protected void configure(HttpSecurity http) throws Exception {
+            // Allow all actuator endpoints.
+            http.requestMatcher(EndpointRequest.toAnyEndpoint())
+                    .authorizeRequests()
+                    .anyRequest()
+                    .permitAll()
+                    .and()
+                    .csrf()
+                    .disable();
       /*http
       .authorizeRequests()
       .anyRequest().permitAll()
       .and()
       //.formLogin();
       .csrf().disable();*/
 
-      // .authenticated().and().httpBasic()..anyRequest().permitAll();
+            // .authenticated().and().httpBasic()..anyRequest().permitAll();
+
+            // Added because of
+            // https://stackoverflow.com/questions/53395200/h2-console-is-not-showing-in-browser
+            http.headers().frameOptions().disable();
+
+        }
+
 
-      // Added because of
-      // https://stackoverflow.com/questions/53395200/h2-console-is-not-showing-in-browser
-      http.headers().frameOptions().disable();
     }
-  }
-}
+
+    @Component
+    @Order(Ordered.HIGHEST_PRECEDENCE)
+    public static class SimpleCorsFilter implements Filter {
```

Comment:
you can also add something like this in any config class -

```java
@Bean
public WebMvcConfigurer corsConfigurer() {
   return new WebMvcConfigurerAdapter() {
      @Override
      public void addCorsMappings(CorsRegistry registry) {
         registry.addMapping("/**")
                      .allowedOrigins("*");
                      /**/
      }    
   };
}
```
Or is there a specific reason for creating a new Filter?

Reply:
Also make this ConditionalOnProperty so we can disable CORS if provided externally (for example by a webserver)

Reply:
Yeah I added the WebMvcConfigurer but it didn't seem to work with Management portal security enabled. It worked when it wasn't enabled.

Reply:
Since we will be using the new library radar-spring-auth for authorisation, this will not be the case anymore and the config should work.

This has been added in https://github.com/RADAR-base/RADAR-Appserver/pull/91. I would suggest to review that first and merge it to dev so this branch can also merge the changes.

Reply:
Thanks the config works now.